### ISSUE TITLE
Project dead?

### ISSUE THREAD

**Issue by imlinus** [xkjyeah/vue-google-maps#514]
Hi,

I'm thinking about using this lib in production. However, 200+ issues is not the best introduction to a library.

Is this project still alive? Or should I consider one of the other options? 

Whats your take guys?

Cheers

**Comment by xkjyeah** [comment]
Hi Linus,

Thanks for asking. I'm not available to actively maintain the project.
Patches are still welcome. Partly also because I'm mostly using MapboxGL at
work instead of Google Maps.

What's urgently needed are:
1. Better automated tests
2. Better integration tests with the popular frameworks, especially Nuxt
and Vue template
3. Better documentation (examples, recommendations)

The above three will go a long way to keeping the project maintainable and
contributable, and will address many of the open issues.

I guess as the project owner, I should have made these things a priority.
Still, I'm somewhat daunted by the challenge of 2) -- I've no idea how best
to spin up a Nuxt project as an automated test and ensure that it works in
dev, ssr and spa mode --, and 3) just requires a lot of time. Do you have
ideas on how to write good documentation? With live code examples?

Part of why it's hard to maintain this project is the absence of good
integration tests. Which means that every change has to be accompanied by
tedious manual tests. Yet every so often, I accidentally break something in
release. All this raises the barriers to entry  to any change or patch.
This something I'm afraid I don't have too much time for.

That said, barring major changes to the Google Maps JS API, there will be
little need to change most parts of the code. Additional properties and
events can be supported with just few additional lines of code to the glue.

**Comment by devWaleed** [comment]
@xkjyeah Brother, look at the weekly npm downloads. So many projects are using this.

I would recommend you to put a bold notice above the readme for contribution. I don't understand test cases, so can't help with that. But I can help with the documentation.

**Comment by xkjyeah** [comment]
@devWaleed  yes i fully understand that the project is widely used. but I'll have to selfishly admit that, if I'm not using it at work, I've very little incentive to bear the pain from my RSI in order to save other projects.

Do you think you can do some documentation with VuePress?

**Comment by pburdette** [comment]
@xkjyeah we use this repo in production on two sites. I think documentation with VuePress is a great idea. I'll get a repo started and a site up for it

**Comment by imlinus** [comment]
Thank you for such a great response.
Trying to keep this repo community-driven is the best way to go.

**Comment by pburdette** [comment]
WIP Docs are live here: https://condescending-lovelace-65c715.netlify.com/

Repo here: https://github.com/beardedpayton/vue-google-maps-docs

**Comment by noobling** [comment]
imo I think it is fine you have provided us with a free service. 

If you want to use this in production to generate income I think its your job to ensure your code is up to scratch not some random guy on the internet who did something for free.

**Comment by imlinus** [comment]
@noobling For sure. I was just checking my alternatives.
I went with my own solution instead as the libraries I found wasn't up to the standards I want.
It's always my job to make sure the code I ship is up to par. That's why I reviewed this library and many others 👍

**Comment by TheAlexLichter** [comment]
Hey :wave: Nuxt core team member here.

first of all, thanks for your work and efforts :clap: 

I've created an example project [here](https://codesandbox.io/s/31j9l75xjm) with Nuxt 2 + `vue2-google-maps`.

Automated testing is possible too, an example can be found in any of my nuxt module repositories (eg [here](https://github.com/Developmint/nuxt-svg-loader/blob/master/test/module.test.js)). I won't have the time to help with actual code though :slightly_frowning_face: 

**But** feel free to drop me a message via Discord (the [Nuxt.js](https://discord.nuxtjs.org/) one or VueLand) if you need further help, I'm quite active there.

**Comment by xkjyeah** [comment]
@manniL Thanks! I've added a link (maybe it's too nondescript?) to the README. Hopefully that helps!

**Comment by mornir** [comment]
Vue.js core team member @akryum is building the same set of components. Wouldn't it be better to converge efforts in one library?
https://github.com/Akryum/vue-googlemaps

**Comment by diegoazh** [comment]
Hi, is someone actually reviewing the opened PRs or granting permissions to merge it? This project is good but it needs more work, if anyone can give me permission on this repo I am willing to do that.
Regards

**Comment by bezborodow** [comment]
> granting permissions to merge

This would be a big help, if somebody trustworthy would volunteer to have higher privileges for this repo. I'd hate to see a multitude of competing forks of this project because contributors do not have simple write access.

**Comment by xkjyeah** [comment]
I'm happy to grant access. But somebody must ask for it first.

**Comment by diegoazh** [comment]
I like to have access to review and merge PRs in this repository if you are in agreement.

**Comment by noobling** [comment]
Just take care on who gets the permissions we don’t want another flatmap to occur.

https://snyk.io/blog/malicious-code-found-in-npm-package-event-stream/

**Comment by diegoazh** [comment]
I use this package in my code, but you have reason @noobling. If you want, take a look on my repos. You decides.

**Comment by xkjyeah** [comment]
@diegoazh I've granted npm access :)

**Comment by bezborodow** [comment]
I'd prefer to see access granted to people [who have contributed in the past](https://github.com/xkjyeah/vue-google-maps/graphs/contributors) (or who even [have pull requests open](https://github.com/xkjyeah/vue-google-maps/pulls)) rather than strangers without any commit history for this project.

**Comment by diegoazh** [comment]
@bezborodow I would like to contribute with this project manteining it, but I don't like to be the only one. It will be great that others contributors will have access too, because other oppinions are always welcome.

**Comment by xkjyeah** [comment]
Hm... alright I'll compromise -- i'll retain npm rights, but I'll let diego (or any other volunteers) accept PRs.

Does that seem reasonable?

**Comment by diegoazh** [comment]
Yes, it's right for me.

**Comment by noobling** [comment]
@JeremiahPinto we should consider maintaining the tools we use for @codersforcauses

**Comment by AllanOcelot** [comment]
Happy to throw my hat into the ring and help maintain things here.
I've used this project multiple times in the past and don't want to see it die off.

**Comment by zyriu** [comment]
Hello, is there any good starting point for adding test cases to the project? I'm a newbie when it comes to front-end development and open-source contributing but I'm going to start using gmaps in production and it would benefit both the project and myself to start contributing.

**Comment by xkjyeah** [comment]
Hey,

Thanks for the interest in contributing!

I don't have a good solution for testing here. Roughly speaking, I imagine
a solution would involve:

- running the tests in a fake browser (e.g. headless chrome)
- mocking google.maps.* objects to some degree of fidelity
- or (bad) loading google.maps.* objects with an API key for testing --
however this makes assertions hard (e.g. asserting that a map is created
when new Map() is called)

In short, the space is open for you to experiment. Feel free to dump the
existing tests.

**Comment by nelsonmfinda** [comment]
> WIP Docs are live here: https://condescending-lovelace-65c715.netlify.com/
> 
> Repo here: https://github.com/beardedpayton/vue-google-maps-docs

Not found

**Comment by tstrohmeier** [comment]
What is the current status of this project?

**Comment by unguul** [comment]
@tstrohmeier The last commit was on February 3rd so definitely not dead.

**Comment by diegoazh** [comment]
No, but the owner did not land a new version, I don't know why 🤷

**Comment by tstrohmeier** [comment]
@unguul Thank you, I saw the commits to but no new versions have been published as also @diegoazh pointed it out now. 

I would suggest this repository is moved to an organization space so that it's not depending on one person.

Is there somewhere the WIP Docs? The repo of @beardedpayton seems to be deleted.

**Comment by unguul** [comment]
@tstrohmeier I'm certain I saw a mention of moving this project to an org space somewhere. Maybe my memory is failing  but either way, that is indeed a good idea.

> Is there somewhere the WIP Docs?

The only docs I'm aware of can be found in the project readme.

Also, as @mornir mentioned, did you notice @Akryium's https://github.com/Akryum/vue-googlemaps ? 

I love this freedom of open source where you can have multiple projects doing the same thing but I also feel like I'm living in an xkcd comic

![standards](https://imgs.xkcd.com/comics/standards.png "Standards")

**Comment by xkjyeah** [comment]
Sigh... 

Sorry folks. I haven't had time to work on this project. However, I've also been cautioned to not simply grant people access to the npm project (which makes sense). I've also granted @diegoazh access to master, but it makes the changes harder to track, personally, and there are some coding decisions that I wanted to keep a control on.

I'd recommend that users who have the bandwidth to maintain this project (and the tons of support questions that people ask) fork and publish their code under a different name. FWIW, this project was vue-google-maps, before I published it as vue2-google-maps on NPM, so there's precedence :).

I've not had the opportunity to use Vue as much these days, but when I return to a fast-paced rapid prototyping kind of work, I hope to see an improved version from the community! Thanks for your support.

**Comment by accexs** [comment]
@xkjyeah Hello, first of all thanks for making this, i completely understand your reasons to dedicate time to other things, hopefully the project can still continue with an active community.

Since you mention your company migrated to **Mapbox** i wanted to ask if it was possible for you to mention differences in price ease of use and potential agains google maps. Thanks!

**Comment by henriquetroiano** [comment]
I´m trying to save some time to contribute with the documentation (better examples)
I´ll make some PR´s as soon as possible.
This is the best maps plugin for VueJS
